stdlib/domain.mli

```diff
@@ -105,6 +105,13 @@ module DLS : sig
         domain will be computed on-demand by the new domain: the first
         [get] call will call the initializer [f] and store that value.
 
+        {b Warning.} [f] may be called several times if another call
+        to [get] occurs during initialization on the same domain. Only
+        the 'first' value computed will be used, the other now-useless
+        values will be discarded. Your initialization function should
+        support this situation, or contain logic to detect this case
+        and fail.
```

Comment:
A late caveat to the guarantee of using the first value: in case an asynchronous exception occurs between computation and assignment, the first computed value will be discarded. Similarly, if an asynchronous exception occurs during computation, then there is no guarantee that the first execution of the initialization function will give the value used eventually, so the logic mentioned can be difficult to implement.

Reply:
Maybe a reasonable view would be to say that the "first thread to compute the value" is defined by the first thread who manages to *set* the value? (This is what we would do in a concurrent setting anyway, where the synchronziation on this final writes and further reads would define the precedence order.)